pkg/provider/provider.go

```diff
@@ -734,25 +753,34 @@ func ConfigureProvider(s *schema.ResourceData) (interface{}, error) {
 		config.DisableTelemetry = v.(bool)
 	}
 
-	if v, ok := s.GetOk("client_request_mfa_token"); ok && v.(bool) {
-		config.ClientRequestMfaToken = gosnowflake.ConfigBoolTrue
+	if v := s.Get("client_request_mfa_token").(string); v != resources.BooleanDefault {
```

Comment:
nit: for the next PR (or even the next one); we already have some helpers in the resources that allow as making such blocks of code much shorter and more readable, it does not differ on the conceptual level from what happens in the resources' create methods. It would be great to extract such methods and reuse them. This logic will be much more compact and much less repetitive then